.github/eval.sh

```diff
@@ -0,0 +1,182 @@
+    echo ""
+}
+
+################################################################################
+
+on_exit () {
+    rm -fr "$TMPDIR" # remove tmpdir
+}
+
+on_err () {
+    kill 0 # kill the master shell and possible subshells
+}
+
+print_err () {
+    printf "${RED}Error:${NONE} %b\n" "$1" >&2
+}
+
+argparse () {
+    # Colorize output if stdout is to a terminal (and not to pipe or file)
+    if [ -t 1 ]; then RED='\033[1;31m'; NONE='\033[0m'; fi
+    # Parse arguments
+    JOB_ID=""; MAX_JOBS=""; EVAL_TARGETS="packages"; OPTIND=1
+    while getopts "hvj:m:t:" copt; do
+        case "${copt}" in
+            h)
+                usage; exit 0 ;;
+            v)
+                set -x ;;
+            t)
+                EVAL_TARGETS="$OPTARG" ;;
+            j)
+                JOB_ID="$OPTARG"
+                if ! [[ "$JOB_ID" == +([0-9]) ]]; then
+                    print_err "'-j' expects a non-negative integer (got: '$JOB_ID')"
+                    usage
+                fi
+                ;;
+            m)
+                MAX_JOBS="$OPTARG"
+                if ! [[ "$MAX_JOBS" == +([0-9]) ]]; then
+                    print_err "'-m' expects a non-negative integer (got: '$MAX_JOBS')"
+                    usage
+                fi
+                ;;
+            *)
+                print_err "unrecognized option"; usage; exit 1 ;;
+        esac
+    done
+    shift $((OPTIND-1))
+    if [ -n "$*" ]; then
+        print_err "unsupported positional argument(s): '$*'"; exit 1
+    fi
+    if [ -z "$JOB_ID" ]; then
+        print_err "missing mandatory option (-j)"; usage; exit 1
+    fi
+    if [ -z "$MAX_JOBS" ]; then
+        print_err "missing mandatory option (-m)"; usage; exit 1
+    fi
+    if [ "$JOB_ID" -ge "$MAX_JOBS" ]; then
+        print_err "'-j' must be smaller than '-m'"; usage; exit 1
+    fi
+}
+
+exit_unless_command_exists () {
+    if ! command -v "$1" &>/dev/null; then
+        print_err "command '$1' is not installed (Hint: are you inside a nix-shell?)"
+        exit 1
+    fi
+}
+
+################################################################################
+
+evaluate () {
+    job="$1"
+    max_jobs="$2"
+    filter="$3"
+    echo "[+] Using filter: '$filter'"
+    # Output all flake output names
+    nix flake show --all-systems --json |\
+      jq  '[paths(scalars) as $path | { ($path|join(".")): getpath($path) }] | add' \
+      >"$TMPDIR/outs_all.txt"
+    # Apply the given filter
+    if ! grep -Po "${filter}\S*.name" "$TMPDIR/outs_all.txt" >"$TMPDIR/outs.txt"; then
+        print_err "No flake outputs match filter: '$filter'"; exit 1
+    fi
+    # Remove the '.name' suffix
+    sed -i "s/.name//" "$TMPDIR/outs.txt"
+    # Read the attribute set names
+    grep -oP ".*(?=\.)" "$TMPDIR/outs.txt" | sort | uniq >"$TMPDIR/attrs.txt"
+    # Generate eval expression on the fly
+    printf '%s\n' \
+      "let" \
+      "  flake = builtins.getFlake ("git+file://" + toString ./.);"\
```

Comment:
Mhm. This will make evaluation impure. I would instead write a new flake.nix into a directory that uses git+file and has flake filtered flake outputs.

Reply:
The eval done here is only to confirm that the evaluation still passes with the changes from the triggering PR. We don't intent to store the evaluation results anywhere.

On building, we run the builds on the r[emote builders with your nix-fast-build](https://github.com/tiiuae/ghaf/blob/489db31b80bc47e114b4a66a1664c095df4f12d4/.github/workflows/build.yml#L141-L146).

Reply:
Ok. I see. In that case impure evaluation wouldn't be a big deal because nix-fast-build would still catch it.